pkgs/applications/misc/smb3-foundry/default.nix

```
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+, makeWrapper
+}:
+let
+  pythonEnv = (python3.withPackages (ps: with ps;[
+    pyside6
+    py65
+    qdarkstyle
+  ]));
+in
+stdenv.mkDerivation rec {
+  pname = "smb3-foundry";
+  version = "1.2";
+
+  src = fetchFromGitHub {
+    owner = "mchlnix";
+    repo = "SMB3-Foundry";
+    rev = "refs/tags/${version}";
+    hash = "sha256-iqqIyGp/sqWgShxk52omVcn7Q3WL2hK8sTLH4dashLE=";
+  };
+
+  patches = [ ./fix-relative-dirs.patch ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
```

Don't forget the runHooks.
```suggestion
  installPhase = ''
    runHook preInstall

```